### ISSUE TITLE
if expression not minified correctly

### ISSUE THREAD

**Issue by adoxa** [putoutjs/minify#12]
- **Minify Version**: v11.2.0
- **Node Version**: v20.13.1
- **OS**: Windows 10 64-bit

This expression:

```javascript
if (var1 || (var2 && var3)) {
   console.log("test")
}
```

incorrectly minifies to:

```javascript
var1||(var2&&var3)&&console.log('test');
```

when it should be:

```javascript
(var1||var2&&var3)&&console.log("test");
```

as it is when using the terser type.

**Comment by coderaiser** [comment]
Thank you! Just fixed and landed in [`v3.12.0`](https://github.com/putoutjs/minify/releases/tag/v3.12.0) 🎉 .

Is it works for you?

**Comment by adoxa** [comment]
The test file works, but the actual code still does not. I'll see if I can come up with another test case later...

**Comment by adoxa** [comment]
Turns out it was just a simple matter of adding another condition:

```javascript
if (var1 || (var2 && var3)) {
   fn1();
   if (var4) {
	   fn2();
   }
}
```

becomes:

```javascript
var1||(var2&&var3)&&(fn1(),var4&&fn2());
```

where the brackets are missing, again.

**Comment by coderaiser** [comment]
Landed in [v3.12.1](https://github.com/putoutjs/minify/releases/tag/v3.12.1) 🎉.

Is it works for you?

**Comment by adoxa** [comment]
Yep, that got it.
